Rebus/Encryption/DecryptMessagesIncomingStep.cs

```diff
@@ -15,18 +15,18 @@ namespace Rebus.Encryption
     [StepDocumentation("Decrypts the body of the incoming message if it has the '" + EncryptionHeaders.ContentEncryption + "' header")]
     public class DecryptMessagesIncomingStep : IIncomingStep
     {
-        readonly IEncryptor _encryptor;
+        readonly IAsyncEncryptor _encryptor;
 
         /// <summary>
         /// Constructs the step with the given encryptor
         /// </summary>
-        public DecryptMessagesIncomingStep(IEncryptor encryptor)
+        public DecryptMessagesIncomingStep(IAsyncEncryptor encryptor)
```

Comment:
I'm somewhat concerned about changing the constructor since this is a public class. It could cause issues if someone is using the `IEncryptor` constructor in their code.

An alternative is keeping the old constructor so that the step has two constructors. I'm not a fan of multiple service constructors, but at least it would be safe

Reply:
Yeah.... well.... while I *lean* towards SemVer compliance, I'm too much of a pragmatic to care about this. 

I would say that it is extremely uncommon for anyone to have taken a compile-time dependency on the signature of this particular constructor.

Let's just push this as it is, and then we'll deal with it later if it causes anyone trouble.